src/contexts/MainContextProvider.jsx

```diff
@@ -0,0 +1,55 @@
+import axios from "axios";
+import { createContext, useContext, useEffect, useReducer } from "react";
+import useAuth from "./AuthContextProvider";
+import reducer from './reducer';
+import ENUMS from '../utils/enums';
+
+const { SET_CART, SET_WISHLIST, SET_PRODUCTS } = ENUMS;
+const MainContext = createContext({});
+
+export const MainContextProvider = ({ children }) => {
```

Comment:
I have a suggestion please change the names MainContext and MainContextProvider to something more meaningful